### PR TITLE
Improve toolchain integration

### PR DESCRIPTION
**Pull request by ehaas**
Work in progress - Add some additional arguments to the linker + path discovery for file components

Next step will be figuring out where GCC is installed

### CODE REVIEW

**Comment by ehaas** [comment]
@Vexu Ok I think this is finally ready for review. There's a lot here, would it be easier if I tried to break it up into a bunch of smaller PRs that build on each other? The logic is mostly adapted from clang; with the caveat that there are probably lots of bugs in the translation because I can't really test it on a wide variety of systems.

File: src/main.zig [hunk 1]
```diff
@@ -3,12 +3,13 @@ const mem = std.mem;
 const Allocator = mem.Allocator;
 const process = std.process;
 const Compilation = @import("Compilation.zig");
-const Driver = @import("Driver.zig");
+const Driver = @import("driver/Driver.zig");
```
Comment (Vexu) [inline src/main.zig:6]: I think the proper naming scheme would be for `Driver.zig` to remain in `src` and the other files related to it being in a `Driver` folder.

File: src/main.zig [hunk 2]
```diff
@@ -3,12 +3,13 @@ const mem = std.mem;
...
 const target_util = @import("target.zig");
+const Toolchain = @import("Toolchain.zig");
 
 var general_purpose_allocator = std.heap.GeneralPurposeAllocator(.{}){};
 
-pub fn main() u8 {
+pub fn main() !u8 {
```
Comment (Vexu) [inline src/main.zig:12]: I'd prefer main to not return an error union.

**Review by Vexu — COMMENTED** [review]
> Ok I think this is finally ready for review. There's a lot here, would it be easier if I tried to break it up into a bunch of smaller PRs that build on each other? The logic is mostly adapted from clang; with the caveat that there are probably lots of bugs in the translation because I can't really test it on a wide variety of systems.

Skimming through I didn't spot anything obviously bad, I'm fine with merging it as is and fixing things as they're found unless you want more throughout reviews?

**Comment by ehaas** [comment]
Ok, made the suggested changes. I'm ok with merging it as-is (assuming it works on your machine) and then fixing issues as they come up when more people start using it on a broader variety of systems.

**Pull request merged**